extensions/osint-web-check/src/UrlInput.tsx

```diff
@@ -0,0 +1,49 @@
+import { Action, ActionPanel, Form } from "@raycast/api";
+import { useState } from "react";
+import { addHttps } from "./utils/addHttps";
+import { checkUrl } from "./utils/checkUrl";
+
+type UrlInputValues = { url: string };
+
+type UrlInputProps = {
+  onSubmit: (url: string) => void;
+};
+
+export function UrlInput({ onSubmit }: UrlInputProps) {
+  const [urlError, setUrlError] = useState<string | undefined>(undefined);
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            onSubmit={async ({ url }: UrlInputValues) => {
+              if (!url) {
+                setUrlError("Enter a URL");
+              }
+
+              url = addHttps(url);
+              if (await checkUrl(url)) {
+                setUrlError(undefined);
+                onSubmit(url);
+              } else {
+                setUrlError("Enter a valid http(s) URL.");
+              }
+            }}
+          />
+        </ActionPanel>
+      }
+    >
+      <Form.Description text="Enter a URL to view OSINT information (DNS, SSL, etc)." />
+      <Form.TextField
+        id="url"
+        title="URL"
+        placeholder="google.com"
```

Comment:
```suggestion
        placeholder={url}
```

Reply:
In this case, `url` will be a *bad* url, so we wouldn't want that as a placeholder. I'm instead using it as an initial value, and opting for a controlled form here – so upon entry here, we'll show the "bad" url they just entered, along with the error message/state.